src/compositor/windowpixmapitem.h

```diff
@@ -36,8 +40,14 @@ class LIPSTICK_EXPORT WindowPixmapItem : public QQuickItem
     Q_PROPERTY(qreal yOffset READ yOffset WRITE setYOffset NOTIFY yOffsetChanged)
     Q_PROPERTY(qreal xScale READ xScale WRITE setXScale NOTIFY xScaleChanged)
     Q_PROPERTY(qreal yScale READ yScale WRITE setYScale NOTIFY yScaleChanged)
+    Q_PROPERTY(int render READ renderOptions WRITE setRenderOptions NOTIFY renderOptionsChanged)
```

Comment:
Either or both.

RenderWindow | RenderSnapshot.  Render the window surface, unless it has been destroyed in which case take a snapshot before releasing the surface and from there on render the snapshot.
RenderWindow: Render the window surface, if it's destroyed render nothing.
RenderSnapshot: Capture a snapshot when a window is hidden or destroyed and from then on render that.   If there is no snapshot render the window surface.

I'm not sold on the API, just getting it working without crashing because something's been released at the wrong time or in the wrong thread has occupied most of my attention so far.  Maybe I'll just register a new type that has the RenderSnapshot behaviour enabled.